HVZ.Web/Shared/Game/TagModal.razor

```diff
@@ -0,0 +1,118 @@
+                                </div>
+                            </div>
+                        </div>
+                    }
+
+                    @if (tagSuccessMsg is not null)
+                    {
+                        <div class="alert alert-success d-flex mt-3">
+                            <div class="d-flex">
+                                <div class="flex-grow-1" id="display-success-msg">
+                                    @tagSuccessMsg
+                                </div>
+                                <div class="flex-grow-0">
+                                    <button class="btn-close" @onclick="ClearMessages"></button>
+                                </div>
+                            </div>
+                        </div>
+                    }
+
+                </div>
+                <div class="modal-footer">
+                    <button class="btn btn-secondary" data-bs-toggle="modal">Close</button>
+                    <button class="btn btn-primary @SubmitEnabledClass" @onclick="LogTag" id="tag-btn">Tag</button>
+                </div>
+            </div>
+        </div>
+    </div>
+}
+
+@code {
+    [Parameter]
+    public Player ActivePlayer { get; set; } = null!;
+
+    [Parameter]
+    public Game ActiveGame { get; set; } = null!;
+
+    string? receiverId;
+
+    string? tagErrorMsg;
+    string? tagSuccessMsg;
+
+    string? SubmitEnabledClass => receiverId?.Length == 4 ? null : "disabled";
+    string? OpenModalEnabledClass => ActiveGame?.IsActive ?? false ? null : "disabled";
+
+    void ClearMessages()
+    {
+        tagErrorMsg = null;
+        tagSuccessMsg = null;
+    }
+
+    async void LogTag()
+    {
+        if (ActiveGame is null)
+            throw new ArgumentNullException("Active game cannot be null");
+
+        if (ActivePlayer is null)
+            throw new ArgumentNullException("Active player cannot be null");
+
+        if (receiverId is null)
+            throw new ArgumentNullException("Make sure the tag button is being properly disabled");
+
+        ClearMessages();
+
+        try
+        {
+            await GameRepo.LogTag(ActiveGame.Id, ActivePlayer.UserId, receiverId);
+            Player targetPlayer = await GameRepo.GetPlayerByGameId(ActiveGame.Id, receiverId);
+            User targetUser = await UserRepo.GetUserById(targetPlayer.UserId);
+
+            tagSuccessMsg = $"You tagged {targetUser.FullName}!";
+        }
+        catch(ArgumentException)
+        {
+            Player? receiver = await GameRepo.FindPlayerByGameId(ActiveGame.Id, receiverId);
+
+            if (receiver is null)
+            {
+                tagErrorMsg = $"{receiverId} is not a valid ID number";
+            }
+
+            else if (receiver.Role != Player.gameRole.Human)
+            {
+                tagErrorMsg = $"The player with the ID: {receiverId} is not a human";
+            }
+        }
```

Comment:
Instead of figuring out what went wrong and creating the error message, we should be able to just use the exception's error message since it _should_ say the same thing.

Reply:
the error messages are formatted differently and contain some information that, while useful as a console error message, is not relevant to the end user

Reply:
thats fine enough then.